libs/geography/src/models/address.ts

```diff
@@ -1,13 +1,40 @@
 /**
- * A basic model of an address
+ * A basic model of an address.
+ * Either a region or region_id must be supplied.
+ * Addresses should either satisfy DaffUnknownRegionAddress or DaffKnownRegionAddress.
  */
 export interface DaffAddress {
 	street: string;
 	street2?: string;
   city: string;
-  region: string;
-  region_id?: string | number;
-  country?: string;
-  country_id?: string | number;
   postcode: string;
-}
\ No newline at end of file
+  country_id: string | number;
+  /**
+   * A field for a region_id supplied by the platform.
+   * Needed for countries that have a list of known regions.
+   * Required if region is not set.
+   */
+  region_id?: string | number;
+  /**
+   * A field for user-supplied regions.
+   * Needed for countries that don't have any regions supplied by the platform.
+   * Required if region_id is not set.
+   */
+  region?: string;
+}
+
+/**
+ * A model representing an address in a country with unknown regions.
+ * The region here should be user-supplied.
+ */
+export interface DaffUnknownRegionAddress extends Omit<DaffAddress, 'region_id'> {
+  region: string;
+}
+
+/**
+ * A model representing an address in a country with a known set of regions with unique IDs.
+ * The region_id should be set to one of those IDs.
+ */
+export interface DaffKnownRegionAddress extends Omit<DaffAddress, 'region'> {
```

Comment:
I think maybe region should still remain in the case that there are some regions defined by the platform, but not all regions are defined.

Reply:
@damienwebdev feel free to chime in but I think thats not a scenario that can happen.

Reply:
Ok, I'm too concerned about it. It would be an obscure and easily fixable bug if it ended up being the case in the future.

Reply:
another question. Can't we just treat the region as the region_id whenever the region_id isn't given by the platform?

Reply:
That way we wouldn't need the region field, and when the user needs to supply a region, we would just write into the region_id field. It would also simplify a lot of this

Reply:
uh haha yes that would be way simpler. But if we do that I vote that we use the `region` field for everything instead of `region_id`.

@damienwebdev Thoughts?